mri/test/test_fourier_adjoint.py

```diff
@@ -155,6 +157,70 @@ def test_NFFT_3D(self):
                 np.testing.assert_allclose(x_d, x_ad, rtol=1e-10)
         print(" NFFT in 3D adjoint test passes")
 
+    def test_adjoint_stack_3D(self):
+        """Test the adjoint operator for the 3D non-Cartesian Fourier transform
+        """
+        for channel in self.num_channels:
+            print("Testing with num_channels=" + str(channel))
+            for i in range(self.max_iter):
+                _mask = np.random.randint(2, size=(self.N, self.N))
+                _mask3D = np.asarray([_mask for i in np.arange(self.N)])
+                _samples = convert_mask_to_locations(_mask3D.swapaxes(0, 2))
+                print("Process Stacked3D-FFT test in 3D '{0}'...", i)
+                fourier_op = Stacked3D(kspace_loc=_samples,
+                                       shape=(self.N, self.N, self.N),
+                                       implementation='cpu', n_coils=channel)
+                Img = np.random.random((channel, self.N, self.N, self.N)) + \
+                    1j * np.random.random((channel, self.N, self.N, self.N))
+                f = np.random.random((channel, _samples.shape[0])) + \
+                    1j * np.random.random((channel, _samples.shape[0]))
+                f_p = fourier_op.op(Img)
+                I_p = fourier_op.adj_op(f)
+                x_d = np.vdot(Img, I_p)
+                x_ad = np.vdot(f_p, f)
+                np.testing.assert_allclose(x_d, x_ad, rtol=1e-10)
+        print("Stacked FFT in 3D adjoint test passes")
+
+    def test_similarity_stack_3D(self):
+        """Test the similarity of stacked implementation of Fourier transform
+        to that of NFFT
+        """
+        for channel in self.num_channels:
+            print("Testing with num_channels=" + str(channel))
+            for N in [64, 128]:
+                # Nz is the number of slices, this would check both N=Nz
+                # and N!=Nz
+                Nz = 64
```

Comment:
Just for computation efficiency could you put a lower value of `Nz` like 2 or 4?

Reply:
Well I need NFFT to also run, and thus I will keep it at Nz=16 (NFFT is unstable at lower Nz). I will test with N=16 and N=32